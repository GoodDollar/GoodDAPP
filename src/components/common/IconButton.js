// @flow
import React from 'react'
import { StyleSheet, Text, TouchableOpacity } from 'react-native'
import Icon from 'react-native-elements/src/icons/Icon'
import normalize from 'react-native-elements/src/helpers/normalizeText'

type IconProps = {
  text: String,
  onPress: Function,
  disabled: Boolean,
  name: String
}

/**
 * Returns a button with an icon and text
 *
 * @param {IconProps} props
 * @param {String} props.text to shown
 * @param {Function} props.onPress action
 * @param {Boolean} props.disabled
 * @param {String} props.name icon name
 * @returns {React.Node}
 */
const IconButton = ({ text, onPress, disabled, ...iconProps }: IconProps) => {
  const styles = createStyles(disabled)
  return (
    <TouchableOpacity style={styles.container} onPress={disabled ? undefined : onPress}>
      <Icon
        size={32}
        reverse
        color="white"
        reverseColor={disabled ? 'rgba(0, 0, 0, 0.32)' : '#282c34'}
        {...iconProps}
      />
      <Text style={styles.text}>{text}</Text>
    </TouchableOpacity>
  )
}

const createStyles = disabled =>
  StyleSheet.create({
    container: {
      flexDirection: 'column',
      alignItems: 'center',
      flex: 1,
      cursor: disabled ? 'inherit' : 'pointer'
    },
    text: {
      color: disabled ? 'rgba(0, 0, 0, 0.32)' : 'inherit',
      fontSize: normalize(11)
    }
  })

export default IconButton
