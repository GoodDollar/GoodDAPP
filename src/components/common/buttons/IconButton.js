// @flow
import React from 'react'
import { TouchableOpacity } from 'react-native'
import Text from '../view/Text'
import { withStyles } from '../../../lib/styles'
import useOnPress from '../../../lib/hooks/useOnPress'
import CustomIcon from './CustomIcon'

type IconProps = {
  text: String,
  onPress: Function,
  disabled: Boolean,
  name: String,
}

/**
 * Returns a button with an icon and text
 *
 * @param {IconProps} props
 * @param {String} props.text to shown
 * @param {Function} props.onPress action
 * @param {Boolean} props.disabled
 * @param {String} props.name icon name
 * @returns {React.Node}
 */
const IconButton = ({ styles, theme, text, onPress, disabled, style, ...iconProps }: IconProps) => {
  const _onPress = useOnPress(onPress)
  return (
    <TouchableOpacity
      cursor={disabled ? 'inherit' : 'pointer'}
      onPress={disabled ? undefined : _onPress}
      style={[styles.container, style]}
    >
      <CustomIcon
        color={theme.colors.darkBlue}
        reverse
        reverseColor={disabled ? 'rgba(0, 0, 0, 0.32)' : '#282c34'}
        size={16}
        circle={32}
        {...iconProps}
      />
      {text && (
        <Text fontSize={11} color={disabled ? 'rgba(0, 0, 0, 0.32)' : ''}>
          {text}
        </Text>
      )}
    </TouchableOpacity>
  )
}

const getStylesFromProps = ({ theme }) => {
  return {
    container: {
      flexDirection: 'row',
      alignItems: 'center',
      display: 'flex',
    },
  }
}

export default withStyles(getStylesFromProps)(IconButton)
