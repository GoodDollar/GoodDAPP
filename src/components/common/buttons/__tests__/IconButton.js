import React from 'react'
import renderer from 'react-test-renderer'
import IconButton from '../IconButton'
import { withThemeProvider } from '../../../../__tests__/__util__'

// Note: test renderer must be required after react-native.
const WrappedComponent = withThemeProvider(IconButton)

describe('IconButton enabled', () => {
  it('matches snapshot', async () => {
    let component

    // eslint-disable-next-line require-await
    await renderer.act(async () => (component = renderer.create(<WrappedComponent text="edit" name="privacy" />)))
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('IconButton disabled', () => {
  it('matches snapshot', async () => {
    let component

    await renderer.act(
      // eslint-disable-next-line require-await
      async () => (component = renderer.create(<WrappedComponent text="edit" name="privacy" disabled />)),
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
