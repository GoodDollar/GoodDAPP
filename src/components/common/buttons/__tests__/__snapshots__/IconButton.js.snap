// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton disabled matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n"
  data-focusable={true}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "MozTransitionProperty": "opacity",
      "MozUserSelect": "none",
      "WebkitAlignItems": "center",
      "WebkitBoxAlign": "center",
      "WebkitBoxDirection": "normal",
      "WebkitBoxFlex": 1,
      "WebkitBoxOrient": "vertical",
      "WebkitFlexBasis": "0%",
      "WebkitFlexDirection": "column",
      "WebkitFlexGrow": 1,
      "WebkitFlexShrink": 1,
      "WebkitTransitionDuration": "0.15s",
      "WebkitTransitionProperty": "opacity",
      "WebkitUserSelect": "none",
      "alignItems": "center",
      "cursor": "pointer",
      "flexBasis": "0%",
      "flexDirection": "column",
      "flexGrow": 1,
      "flexShrink": 1,
      "msFlexAlign": "center",
      "msFlexDirection": "column",
      "msFlexNegative": 1,
      "msFlexPositive": 1,
      "msFlexPreferredSize": "0%",
      "msTouchAction": "manipulation",
      "msUserSelect": "none",
      "touchAction": "manipulation",
      "transitionDuration": "0.15s",
      "transitionProperty": "opacity",
      "userSelect": "none",
    }
  }
  tabIndex="0"
>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "WebkitAlignItems": "center",
        "WebkitBoxAlign": "center",
        "WebkitBoxPack": "center",
        "WebkitJustifyContent": "center",
        "alignItems": "center",
        "borderBottomLeftRadius": "16px",
        "borderBottomRightRadius": "16px",
        "borderTopLeftRadius": "16px",
        "borderTopRightRadius": "16px",
        "height": "32px",
        "justifyContent": "center",
        "msFlexAlign": "center",
        "msFlexPack": "center",
        "width": "32px",
      }
    }
  >
    <div
      className="css-text-901oao"
      dir="auto"
      style={
        Object {
          "color": "rgba(255,255,255,1.00)",
          "fontFamily": "gooddollar",
          "fontSize": "16px",
          "fontStyle": "normal",
          "fontWeight": "normal",
        }
      }
    >
      
    </div>
  </div>
  <div
    className="css-text-901oao"
    dir="auto"
    style={
      Object {
        "color": "rgba(0,0,0,0.32)",
        "direction": "ltr",
        "fontFamily": "Roboto",
        "fontSize": "11px",
        "fontWeight": "normal",
        "lineHeight": "13.2px",
        "textAlign": "center",
        "textDecoration": "none",
        "textTransform": "none",
      }
    }
  >
    edit
  </div>
</div>
`;

exports[`IconButton enabled matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n"
  data-focusable={true}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "MozTransitionProperty": "opacity",
      "MozUserSelect": "none",
      "WebkitAlignItems": "center",
      "WebkitBoxAlign": "center",
      "WebkitBoxDirection": "normal",
      "WebkitBoxFlex": 1,
      "WebkitBoxOrient": "vertical",
      "WebkitFlexBasis": "0%",
      "WebkitFlexDirection": "column",
      "WebkitFlexGrow": 1,
      "WebkitFlexShrink": 1,
      "WebkitTransitionDuration": "0.15s",
      "WebkitTransitionProperty": "opacity",
      "WebkitUserSelect": "none",
      "alignItems": "center",
      "cursor": "pointer",
      "flexBasis": "0%",
      "flexDirection": "column",
      "flexGrow": 1,
      "flexShrink": 1,
      "msFlexAlign": "center",
      "msFlexDirection": "column",
      "msFlexNegative": 1,
      "msFlexPositive": 1,
      "msFlexPreferredSize": "0%",
      "msTouchAction": "manipulation",
      "msUserSelect": "none",
      "touchAction": "manipulation",
      "transitionDuration": "0.15s",
      "transitionProperty": "opacity",
      "userSelect": "none",
    }
  }
  tabIndex="0"
>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "WebkitAlignItems": "center",
        "WebkitBoxAlign": "center",
        "WebkitBoxPack": "center",
        "WebkitJustifyContent": "center",
        "alignItems": "center",
        "borderBottomLeftRadius": "16px",
        "borderBottomRightRadius": "16px",
        "borderTopLeftRadius": "16px",
        "borderTopRightRadius": "16px",
        "height": "32px",
        "justifyContent": "center",
        "msFlexAlign": "center",
        "msFlexPack": "center",
        "width": "32px",
      }
    }
  >
    <div
      className="css-text-901oao"
      dir="auto"
      style={
        Object {
          "color": "rgba(255,255,255,1.00)",
          "fontFamily": "gooddollar",
          "fontSize": "16px",
          "fontStyle": "normal",
          "fontWeight": "normal",
        }
      }
    >
      
    </div>
  </div>
  <div
    className="css-text-901oao"
    dir="auto"
    style={
      Object {
        "color": "inherit",
        "direction": "ltr",
        "fontFamily": "Roboto",
        "fontSize": "11px",
        "fontWeight": "normal",
        "lineHeight": "13.2px",
        "textAlign": "center",
        "textDecoration": "none",
        "textTransform": "none",
      }
    }
  >
    edit
  </div>
</div>
`;
